core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -76,8 +76,14 @@ public void put(PageId pageId, byte[] page) throws IOException {
       Files.createDirectories(parent);
       Files.createFile(p);
     }
-    try (FileOutputStream fos = new FileOutputStream(p.toFile(), false)) {
-      fos.write(page);
+    try {
+      // extra try to ensure output stream is closed
+      try (FileOutputStream fos = new FileOutputStream(p.toFile(), false)) {
+        fos.write(page);
+      }
+    } catch (IOException e) {
```

Comment:
Should we catch throwable here?

Reply:
changed to `Exception`. If `Error` happens, that's something really bad and we should escalate.